GPT/gpt.py

```diff
@@ -192,21 +195,28 @@ def gpt_reformat_last(how_to_reformat: str):
             notify("No text to reformat")
             raise Exception("No text to reformat")
 
+    def gpt_paste_with_modifiers(text: str, modifiers: list[str] = ""):
+        """Paste and apply modifications"""
+        for modifier in modifiers:
+            match modifier:
+                case "above":
+                    actions.edit.line_insert_up()
+                case "below":
+                    actions.edit.line_insert_down()
+                case "snip":
+                    actions.user.insert_snippet(text)
+                    return
+
+        actions.user.paste(text)
+
     def gpt_insert_response(
-        result: str, method: str = "", cursorless_destination: Any = None
+        result: str,
+        modifier: list[str] = [""],
+        method: str = "",
+        cursorless_destination: Any = None,
     ):
         """Insert a GPT result in a specified way"""
         match method:
-            case "above":
-                actions.key("left")
-                actions.edit.line_insert_up()
-                actions.user.paste(result)
-                GPTState.last_was_pasted = True
-            case "below":
-                actions.key("right")
-                actions.edit.line_insert_down()
-                actions.user.paste(result)
-                GPTState.last_was_pasted = True
```

Comment:
We should keep GPTState info. We still want to keep track of things in case they want to select it after the fact. Most users will select after the fact, not in the request. If we get rid of this variable we don't know if the last response can be pasted.

Reply:
I'm still tracking it below in the default case. It will be set for above and below as well as for snippets. Is that still a problem?